src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp

```diff
@@ -0,0 +1,4587 @@
+    {
+        HRESULT gHR = S_OK;
+
+        //
+        // Continuous Profiler
+        //
+        if (m_cpProfiler != nullptr)
+        {
+            HRESULT hr = m_cpProfiler->EventPipeEventDelivered(
+                provider, eventId, eventVersion, cbMetadataBlob, metadataBlob, cbEventData, eventData, pActivityId,
+                pRelatedActivityId, eventThread, numStackFrames, stackFrames);
+            if (FAILED(hr))
+            {
+                Warn("[Continuous Profiler] Error in EventPipeEventDelivered() call.");
+                gHR = hr;
+            }
+        }
+
+        //
+        // Tracer
+        //
+        if (m_tracerProfiler != nullptr)
+        {
+            HRESULT hr = m_tracerProfiler->EventPipeEventDelivered(
+                provider, eventId, eventVersion, cbMetadataBlob, metadataBlob, cbEventData, eventData, pActivityId,
+                pRelatedActivityId, eventThread, numStackFrames, stackFrames);
+            if (FAILED(hr))
+            {
+                Warn("[Tracer] Error in EventPipeEventDelivered() call.");
+                gHR = hr;
+            }
+        }
+
+        //
+        // Custom
+        //
+        if (m_customProfiler != nullptr)
+        {
+            HRESULT hr = m_customProfiler->EventPipeEventDelivered(
+                provider, eventId, eventVersion, cbMetadataBlob, metadataBlob, cbEventData, eventData, pActivityId,
+                pRelatedActivityId, eventThread, numStackFrames, stackFrames);
+            if (FAILED(hr))
+            {
+                Warn("[Custom] Error in EventPipeEventDelivered() call.");
+                gHR = hr;
+            }
+        }
+
+        return gHR;
+    }
+
+    HRESULT STDMETHODCALLTYPE CorProfiler::EventPipeProviderCreated(EVENTPIPE_PROVIDER provider)
+    {
+        HRESULT gHR = S_OK;
+
+        //
+        // Continuous Profiler
+        //
+        if (m_cpProfiler != nullptr)
+        {
+            HRESULT hr = m_cpProfiler->EventPipeProviderCreated(provider);
+            if (FAILED(hr))
+            {
+                Warn("[Continuous Profiler] Error in EventPipeProviderCreated() call.");
+                gHR = hr;
+            }
+        }
+
+        //
+        // Tracer
+        //
+        if (m_tracerProfiler != nullptr)
+        {
+            HRESULT hr = m_tracerProfiler->EventPipeProviderCreated(provider);
+            if (FAILED(hr))
+            {
+                Warn("[Tracer] Error in EventPipeProviderCreated() call.");
+                gHR = hr;
+            }
+        }
+
+        //
+        // Custom
+        //
+        if (m_customProfiler != nullptr)
+        {
+            HRESULT hr = m_customProfiler->EventPipeProviderCreated(provider);
+            if (FAILED(hr))
+            {
+                Warn("[Custom] Error in EventPipeProviderCreated() call.");
+                gHR = hr;
+            }
+        }
+
+        return gHR;
```

Comment:
From what I can tell, they are all identical. If I missed a callback that is different structure from others, please tell me which. :)

WOuld it make sense to shorten this with some soft of compile-time expanded template / macro?

Reply:
done in 4e9238c